[BUG] Inconsistently crashing when work profile is locked

### Describe the bug

Occasionally, the launcher will crash when the work profile is locked. The stack trace notes `User 10 must be unlocked for widgets to be available`, which made me suspect that one of the widgets I created used the work profile. Removing all of the widgets and restarting the app still caused it to crash. Unlocking the work profile stops the launcher from crashing.

### Steps to reproduce

I haven't been able to manually reproduce the bug. I captured these logs when the issue was occurring.

### Expected behavior

When the work profile is locked, the launcher shouldn't crash.

### Screenshots

_No response_

### Device information

- Device: Google Pixel 10 Pro XL
- OS: Android 16 (CP1A.260305.018)

### App version

15.Dev.(#3248)

### Additional context

```
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: FATAL EXCEPTION: launcher-loader
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: Process: app.lawnchair.nightly, PID: 29013
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: java.lang.IllegalStateException: User 10 must be unlocked for widgets to be available
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3378)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3354)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3337)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3279)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.internal.appwidget.IAppWidgetService$Stub$Proxy.getWidgetPreview(IAppWidgetService.java:1412)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.appwidget.AppWidgetManager.getWidgetPreview(AppWidgetManager.java:1625)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.launcher3.widget.WidgetManagerHelper.loadGeneratedPreview(r8-map-id-e33d4cc7963552f2a7a771e8110415a49d167b08a195864569173a9865b578e3:9)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.launcher3.model.WidgetItem.<init>(r8-map-id-e33d4cc7963552f2a7a771e8110415a49d167b08a195864569173a9865b578e3:14)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.launcher3.model.WidgetsModel.update(r8-map-id-e33d4cc7963552f2a7a771e8110415a49d167b08a195864569173a9865b578e3:69)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.launcher3.model.LoaderTask.run(r8-map-id-e33d4cc7963552f2a7a771e8110415a49d167b08a195864569173a9865b578e3:368)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:1095)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Handler.dispatchMessageImpl(Handler.java:135)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:125)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:269)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:367)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:139)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.server.pm.ShortcutService$$ExternalSyntheticBUOutline0.m(R8$$SyntheticClass:0)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.server.appwidget.AppWidgetServiceImpl.ensureGroupStateLoadedLocked(AppWidgetServiceImpl.java:1199)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.server.appwidget.AppWidgetServiceImpl.ensureGroupStateLoadedLocked(AppWidgetServiceImpl.java:1184)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.server.appwidget.AppWidgetServiceImpl.getWidgetPreview(AppWidgetServiceImpl.java:5285)
03-20 14:29:20.265 10307 29013 29035 E AndroidRuntime: 	at com.android.internal.appwidget.IAppWidgetService$Stub.onTransact(IAppWidgetService.java:755)
```

--- a/src/com/android/launcher3/widget/WidgetManagerHelper.java
+++ b/src/com/android/launcher3/widget/WidgetManagerHelper.java
@@ -174,6 +174,9 @@ public class WidgetManagerHelper {
         } catch (NoSuchMethodError | NoClassDefFoundError e) {
             Log.w("LC_"+TAG, "loadGeneratedPreview: Error loading widget preview");
             return null;
+        } catch (IllegalStateException e) {
+            Log.w("LC_"+TAG, "loadGeneratedPreview: Invalid state fetching preview.", e);
+            return null;
         }
     }
 

--- a/tests/multivalentTests/src/com/android/launcher3/widget/WidgetManagerHelperTest.kt
+++ b/tests/multivalentTests/src/com/android/launcher3/widget/WidgetManagerHelperTest.kt
@@ -125,4 +125,16 @@ class WidgetManagerHelperTest {
             verify(appWidgetManager).getWidgetPreview(provider, profile, widgetCategory)
         }
     }
+
+    @Test
+    fun loadGeneratedPreview_returnsNull_whenAppWidgetManagerThrowsIllegalStateException() {
+        val widgetCategory = 130
+
+        whenever(appWidgetManager.getWidgetPreview(info.provider, info.profile, widgetCategory))
+            .thenThrow(IllegalStateException("User must be unlocked for widgets to be available"))
+
+        val result = underTest.loadGeneratedPreview(info, widgetCategory)
+
+        Truth.assertThat(result).isNull()
+    }
 }

